rasterio/rio/edit_info.py

```diff
@@ -94,10 +152,19 @@ def transform_handler(ctx, param, value):
 @click.option('--all', 'allmd', callback=all_handler, flag_value='like',
               is_eager=True, default=False,
               help="Copy all metadata items from the template file.")
+@click.option(
+    '--colorinterp', callback=colorinterp_handler,
+    metavar="bidx=ci[,bidx=ci,...]|RGB|RGBA|like",
+    help="Set color interpretation for one or more bands.  Can also use "
+         "'RGBA' as shorthand for '1=red,2=green,3=blue,4=alpha' and 'RGB' "
+         "for the same sans alpha band.  This cannot be combined with "
+         "individual band definitions.  Use 'like' to inherit color "
+         "interpretation from '--like'.  Results may be unexpected if the "
+         "input image and template image have a differing number of bands.")
```

Comment:
Should we raise a warning or an exception in this case?  I'm having a hard time thinking of a case where we'd want to apply color interpretation between two rasters with similar numbers of bands: the only case possibly being going from RGBA to RGB (4 to 3 bands).